enhancements/cluster-logging/cluster-logging-es-pipeline-processing.md

```diff
@@ -0,0 +1,205 @@
+
+- Approach C) is on the other side of the spectrum. All data is stored into common index and whenever user needs to store data with a new data model then it is necessary to transform the data to fit existing data model in existing index. This approach requires attention to data model and users may need to invest time and resources to transform new data sources. On the other hand there is minimum additional indices created in ES cluster and data can be queried uniformly because document fields have predictable meaning and types.
+
+It should be pointed out that Elasticsearch pipeline processes can support both cases. Depending on how much users are willing to pay attention to their data model they can start with approach B) and ventually transition to approach C) once they feel more confident about managing their own data model across more log resources.
+
+In this proposal we assume that the approach B is the one we would start enabling first. But nothing is stopping us from enabling the approach C in the future. Though it is fair to say that approach C will be more complicated to implement because it will require management of custom processing pipelines.
+
+We will be successful when:
+
+* An OCP user (or admin only?)  is able to configure specific logs to be pointed into different (non-default) index in Elasticsearch based on the data contained in the document itself.
+* An OCP user (or admin only?) has enough information and tooling to undertstand and evaluate the impact of creating additional indices. This specifically means new index level metrics will need to be introduced and relevant alerts must be provided as well. Notice that we have recently turned off index level Prometheus metrics (because they were high-cardinality metrics) thus we will need to revisit index level metrics and provide new aggregated non-high cardinality metrics.
+* Any additional non-default indices must be subject to existing retention policy. At the implementation level this means that any new indices have names that comply with data retention implementation OOTB.
+* A multi-tenant data visibility works seamlessly.
+* A roll-over process can manage additional indices seamlessly.
+* An OCP user is able to query data from additional indices.
+* An OCP user is provided enough documentation to be able to understand **when** and **how** to use this feature.
+
+Optional points, not required for inifial implementation:
+
+* An OCP user (or admin only?) can provide custom index mapping for the data. This is specifically needed if the [default Elasticsearch type inference mechanism](https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic-field-mapping.html) does not lead to acceptable results.
```

Comment:
so, this is pretty different from what we currently allow. how would you propose we expose a mechanism to do this? is there an easy way we can safeguard from someone breaking the cluster at the same time?

Reply:
This is optional point and I think Christian did not want to address this in first iteration (hence we moved this to "optional"). How this is implemented can be tightly related to how we decide who is responsible for creating a new index (users vs. admins). But I can imagine we can enable some mean of pushing new index templates into ES where they must comply with project or namespace naming schema (so that users can not override index templates for projects they do not have access to.

Other that this this can give users more power and they must behave responsibly. And this is where we are opening something that will be important for successful implementation of approach C) anyway. We need to see if this is really possible or not. I believe if users are educated and they learn importance of index mapping then they should be able to get there. If not then this can be dangerous.